activerecord/lib/rails/generators/active_record/model/model_generator.rb

```diff
@@ -48,7 +48,7 @@ def generate_application_record
 
         # Used by the migration template to determine the parent name of the model
         def parent_class_name
-          options[:parent] || determine_default_parent_class
+          options[:parent] || 'ApplicationRecord'
```

Comment:
Yup. Don't need to freeze strings. We only do in critical paths

On Wed, Apr 27, 2016, 14:01 Genadi Samokovarov <notifications@github.com>
wrote:

> In
> activerecord/lib/rails/generators/active_record/model/model_generator.rb
> <https://github.com/rails/rails/pull/24756#discussion_r61296216>:
>
> > @@ -48,7 +48,7 @@ def generate_application_record
> >
> >          # Used by the migration template to determine the parent name of the model
> >          def parent_class_name
> > -          options[:parent] || determine_default_parent_class
> > +          options[:parent] || 'ApplicationRecord'
>
> Do we have a general rule of thumb for freezing strings?
>
> This is run once in a generator, that will exit in a reasonable finite
> time. I don't see how much allocations or performance can we squeeze out of
> this single run.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/rails/pull/24756/files/a33f6453bf1c01d7d78e2be10b78117f341761d5#r61296216>
>